### PR TITLE
Lwm2m: add more tests

### PR DESCRIPTION
**Pull request by SeppoTakalo**
Add more tests to
* RD Client state machine
* Engine

Refactor engine, so that utility functions are moved to lwm2m_util.c

### CODE REVIEW

**Review by rlubos — APPROVED** [review]

File: subsys/net/lib/lwm2m/lwm2m_util.c [hunk 1]
```diff
@@ -528,3 +528,51 @@ bool lwm2m_obj_path_equal(const struct lwm2m_obj_path *a, const struct lwm2m_obj
 
 	return true;
 }
+
+/* for debugging: to print IP addresses */
+char *lwm2m_sprint_ip_addr(const struct sockaddr *addr)
```
Comment (jukkar) [inline subsys/net/lib/lwm2m/lwm2m_util.c:533]: We have similar functions in core net although they are now "private" and not in public api. I just wonder if we should create for example `<net/utils.h>` that could contain various helper functions so that these would not be needed to implement in the network libraries and/or applications. This is just brainstorming and not part of this PR.

File: tests/net/lib/lwm2m/lwm2m_rd_client/src/main.c [hunk 2]
```diff
@@ -244,7 +288,36 @@ ZTEST(lwm2m_rd_client, test_start_registration_update)
...
+	lwm2m_init_message_fake.custom_fake = lwm2m_init_message_fake_default;
+	coap_header_get_code_fake.custom_fake = coap_header_get_code_fake_created;
+	coap_find_options_fake.custom_fake = coap_find_options_do_registration_reply_cb_ok;
+	zassert_true(lwm2m_rd_client_start(&ctx, "Test", 0, lwm2m_event_cb, lwm2m_observe_cb) == 0,
+		     NULL);
+	zassert_true(check_lwm2m_rd_client_event(LWM2M_RD_CLIENT_EVENT_REGISTRATION_COMPLETE, 0),
+		     NULL);
+
+	engine_update_tx_time();
+	k_sleep(K_SECONDS(15));
```
Comment (jukkar) [inline tests/net/lib/lwm2m/lwm2m_rd_client/src/main.c:319]: Do we really need such a long sleep here, seems quite excessive

Comment (SeppoTakalo) [reply tests/net/lib/lwm2m/lwm2m_rd_client/src/main.c:319]: This test runs on `native_posix` target and does not require "-rt" flag, so it is not effectively 15 seconds in CI.

Example on my machine
```
------ TESTSUITE SUMMARY START ------

SUITE PASS - 100.00% [lwm2m_rd_client]: pass = 13, fail = 0, skip = 0, total = 13 duration = 56.610 seconds
...
------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION SUCCESSFUL
west build -t run  0,22s user 0,03s system 104% cpu 0,243 total
```

But yes, we need a longer sleep here, because in settings 10 second is the RX window, and only after that we receive the RX_OFF event.

**Review by jukkar — APPROVED** [review]
